Gemfile

```diff
@@ -20,6 +20,7 @@ gem "gds-sso", "12.1.0"
 gem 'bunny', '2.5.1'
 gem 'whenever', '0.9.4', require: false
 gem "govuk_sidekiq", "~> 0.0"
+gem 'sidekiq', git: "https://github.com/alphagov/sidekiq.git", branch: "fix-testing-with-overridden-queue"
```

Comment:
This change is now in Sidekiq master (https://github.com/mperham/sidekiq/pull/3152), and I'll keep an eye out for releases.